eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelEvents.scala

```diff
@@ -38,7 +38,7 @@ case class ChannelIdAssigned(channel: ActorRef, remoteNodeId: PublicKey, tempora
 
 case class ShortChannelIdAssigned(channel: ActorRef, channelId: ByteVector32, shortChannelId: ShortChannelId, previousShortChannelId: Option[ShortChannelId]) extends ChannelEvent
 
-case class LocalChannelUpdate(channel: ActorRef, channelId: ByteVector32, shortChannelId: ShortChannelId, remoteNodeId: PublicKey, channelAnnouncement_opt: Option[ChannelAnnouncement], channelUpdate: ChannelUpdate, previousChannelUpdate_opt: Option[ChannelUpdate], commitments: AbstractCommitments) extends ChannelEvent
+case class LocalChannelUpdate(channel: ActorRef, channelId: ByteVector32, shortChannelId: ShortChannelId, remoteNodeId: PublicKey, channelAnnouncement_opt: Option[ChannelAnnouncement], channelUpdate: ChannelUpdate, hasChanged: Boolean, commitments: AbstractCommitments) extends ChannelEvent
```

Comment:
This is a very confusing name, because the `commitments` object may have changed greatly while the `channelUpdate` didn't, so this will be false in cases where important data inside this class has changed, downstream components may misinterpret this field.

Reply:
I've reverted to including the full previous channel update.